### ISSUE TITLE
[aws-eks] ‼️ NOTICE: FargateCluster and KubernetesPatch breaks deployment

### ISSUE THREAD

**Issue by iliapolo** [aws/aws-cdk#10537]
Please add your +1 👍 to let us know you have encountered this

---

### Overview:

Starting from version `1.62.0`, the following scenarios are unable to deploy:

#### Fargate Cluster

```typescript
import * as eks from '@aws-cdk/aws-eks';
import * as cdk from '@aws-cdk/core';

export class EksTestStack extends cdk.Stack {
  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const cluster = new eks.FargateCluster(this, 'cluster', {
      version: eks.KubernetesVersion.V1_17,
    })
  }
}
```

#### Kubernetes Patch

```typescript
import * as eks from '@aws-cdk/aws-eks';
import * as cdk from '@aws-cdk/core';

export class EksTestStack extends cdk.Stack {
  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const cluster = new eks.Cluster(this, 'cluster', {
      version: eks.KubernetesVersion.V1_17,
    })

    new eks.KubernetesPatch(this, 'Patch', {...});
  }
}
```

#### Complete Error Message: 

`Circular dependency between resources: [...]`

### Workaround:

No workaround at the moment.

### Solution:

We've released a patch: [`1.64.1`](https://github.com/aws/aws-cdk/releases/tag/v1.64.1)

### Related Issues:

Originally reported in https://github.com/aws/aws-cdk/issues/10528

**Comment by iliapolo** [comment]
Re-opening until we release a version that includes the fix

**Comment by iliapolo** [comment]
Issue was resolved and the fix is now part of [1.64.1](https://github.com/aws/aws-cdk/releases/tag/v1.64.1)
